libyul/optimiser/ControlFlowSimplifier.cpp

```diff
@@ -205,17 +203,17 @@ OptionalStatements ControlFlowSimplifier::reduceSingleCaseSwitch(Switch& _switch
 	yulAssert(_switchStmt.cases.size() == 1, "Expected only one case!");
 
 	auto& switchCase = _switchStmt.cases.front();
-	auto loc = locationOf(*_switchStmt.expression);
+	shared_ptr<DebugData const> debugData = debugDataOf(*_switchStmt.expression);
 	YulString type = m_typeInfo.typeOf(*_switchStmt.expression);
 	if (switchCase.value)
 	{
 		if (!m_dialect.equalityFunction(type))
 			return {};
 		return make_vector<Statement>(If{
-			std::move(_switchStmt.location),
+			std::move(_switchStmt.debugData),
```

Comment:
Is this move useful?

Reply:
Yes, it's a shared pointer, they should always be  moved.

Reply:
Most of them are not moved, I only commented on the cases which were moved. Need to change then.

Reply:
The ones that were moved before are good to move now. For the others, I would not do it in this PR because it needs a reference analysis.